language-design.md

```diff
@@ -49,7 +86,11 @@ In what way is an API a language?
 
 **Response**
 
+APIs are actually a lot like DSLs, especially internal ones. In fact, they actually match Fowler's definition of a DSL quite well [Fowler, p 27-28]. They are used by people to give instructions to computers. Good APIs should also have the fluency and feel of a programming languages. This is backed up by Joshua Bloch, who wrote "strive for intelligibility, consistency, and symmetry [in APIs]. Every API is a little language, and people must learn to read and write it. If you get an API right, code will read like prose" [Bloch, 2006, p 1]. APIs generally have limited expressiveness. Even if they are pretty flexible, APIs usually to carry out their operations through another piece of software or on another computer, which is kind of a limited domain in and of itself. I think it would follow that APIs will also have the domain focus that Fowler mentions as his last characteristic of a DSL. The whole purpose of using an API is because it's better at what you're trying to than the GPPL that it's implemented in.
 
```

Comment:
I like the comparison between APIs and DSLs a lot, but it leaves me with the question of what the difference is between them. Like you said, APIs make single tasks easier, and have limited scope, which is the entire point of a DSL. Certainly its the case that APIs run on top of an existing framework, but some DSLs do as well. So does this mean APIs are a type of DSL? Fowler has some points about this,  but the line seems to be more blurry than I had previously thought.